qiskit/providers/ibmq/ibmqbackend.py

```diff
@@ -289,6 +289,14 @@ def run(
                 "'use_measure_esp' is unset or set to 'False'."
             )
 
+        if run_config.get("dynamic", None):
```

Comment:
Is your intent to deliberately permit setting `dynamic=False` without an error?  If so, then fine, if not, maybe you wanted `if run_config.get("dynamic", None) is not None:`.

Reply:
Initially this is my intent since this will pass-through unset, but on second thought I'll just prevent any setting of this value.

Reply:
Updated in 57dcc2d.